streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -335,6 +342,42 @@ private static <R> QueryResult<R> runWindowRangeQuery(final Query<R> query,
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private static <R> QueryResult<R> runVersionedKeyQuery(final Query<R> query,
+                                                           final PositionBound positionBound,
+                                                           final QueryConfig config,
+                                                           final StateStore store) {
+        if (store instanceof VersionedKeyValueStore) {
+            final VersionedKeyValueStore<Bytes, byte[]> versionedKeyValueStore =
+                (VersionedKeyValueStore<Bytes, byte[]>) store;
+            if (query instanceof VersionedKeyQuery) {
```

Comment:
Same as above -- I think we should remove this check, as the code guarantees that it's true.

Reply:
> Same as above -- I think we should remove this check, as the code guarantees that it's true.

here I agree with you.